atlasdb-config/src/main/java/com/palantir/atlasdb/factory/PersistentStorageFactory.java

```diff
@@ -17,14 +17,15 @@
 package com.palantir.atlasdb.factory;
 
 import com.palantir.atlasdb.config.RocksDbPersistentStorageConfig;
-import com.palantir.atlasdb.persistent.api.PersistentTimestampStore;
+import com.palantir.atlasdb.persistent.api.PersistentStore;
+import com.palantir.atlasdb.persistent.rocksdb.RocksDbPersistentStore;
 
 public interface PersistentStorageFactory {
     /**
-     * Constructs a {@link com.palantir.atlasdb.persistent.rocksdb.RocksDbPersistentTimestampStore} using the
-     * supplied configuration.
+     * Constructs a {@link RocksDbPersistentStore} using the supplied configuration.
+     *
      * @param config to use to configure the store
      * @return store to be used
      */
-    PersistentTimestampStore constructPersistentTimestampStore(RocksDbPersistentStorageConfig config);
+    PersistentStore constructPersistentStore(RocksDbPersistentStorageConfig config);
```

Comment:
As an aside, this is weird. Everything is quite generic until it isn't. Similar to the async c* stuff, this should be a `PersistentStorageConfig`.

Reply:
This was intentional since there is only one config so this is to keep it simple. Can be address in following PRs.